src/configure/helper/templateHelper.ts

```diff
@@ -112,35 +139,50 @@ export async function analyzeRepoAndListAppropriatePipeline(repoPath: string, re
     templateResult = targetResource && !!targetResource.kind ? templateResult.filter((template) => !template.targetKind || template.targetKind.toLowerCase() === targetResource.kind.toLowerCase()) : templateResult;
     templateResult = templateResult.filter((pipelineTemplate) => pipelineTemplate.enabled);
 
-    // remove duplicate named template:
-    templateResult = removeDuplicates(templateResult);
     return templateResult;
-
 }
 
-export async function analyzeRepoAndListAppropriatePipeline2(azureSession: AzureSession, repoPath: string, repositoryProvider: RepositoryProvider, repoAnalysisParameters: RepositoryAnalysisParameters, targetResource?: GenericResource): Promise<PipelineTemplateMetadata[]> {
+export async function analyzeRepoAndListAppropriatePipeline2(azureSession: AzureSession, repoPath: string, repositoryProvider: RepositoryProvider, repoAnalysisParameters: RepositoryAnalysisParameters, targetResource?: GenericResource): Promise<PipelineTemplate[]> {
 
-    //TO:DO - Merge local repo analysis (Some changes in the definition of AnalysisResult required)
-    let templateResult: PipelineTemplateMetadata[] = [];
+    var pipelineTemplates: PipelineTemplate[] = [];
+    var localPipelineTemplatesPromise: Promise<LocalPipelineTemplate> = await this.analyzeRepoAndListAppropriatePipeline(repoPath, repositoryProvider, repoAnalysisParameters);
 
     let serviceClient = new TemplateServiceClient(azureSession.credentials);
-    templateResult = await serviceClient.getTemplates(repoAnalysisParameters);
-    templateResult = templateResult.sort((a, b) => {
-        if (a.templateWeight > b.templateWeight) { return 1; }
-        else { return -1; }
-    });
-    return templateResult;
+    var templateResultPromise = await serviceClient.getTemplates(repoAnalysisParameters);
```

Comment:
getTemplates from TemplateService only if repoAnalysisParameters is not null and contains repositoryAnalysisApplicationSettingsList

Reply:
updated